bcache-tools doesn't cross-compile

This is pulled in unconditionally by `nixos/modules/tasks/bcache.nix` and fails during NixOS cross-compilation:

```
unpacking sources
unpacking source archive /nix/store/rgwm01hfhmbba4nbfbayimb81v9wqrnb-bcache-tools-1.0.7.tar.gz
source root is bcache-tools-1.0.7
setting SOURCE_DATE_EPOCH to timestamp 1399729101 of file bcache-tools-1.0.7/probe-bcache.c
patching sources
applying patch /nix/store/2q3z7587yhlz0i2xvfvvap42zk5carlv-bcache-udev-modern.patch
patching file 69-bcache.rules
patching file Makefile
Hunk #1 succeeded at 9 with fuzz 1.
patching file bcache-register
applying patch /nix/store/03sl46khd8gmjpsad7223m32ma965vy9-fix-static.patch
patching file bcache.c
updateAutotoolsGnuConfigScriptsPhase
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/n313xks5ym0s0a5v8a5285rmnmvy6ms9-bash-4.4-p23/bin/bash PREFIX=\"/nix/store/49rwsqj9h0r5sw85p1yyhf9nrm5c5xd2-bcache-tools-1.0.7-armv7l-unknown-linux-gnueabihf\" UDEVLIBDIR=\"/nix/store/49rwsqj9h0r5sw85p1yyhf9nrm5c5xd2-bcache-tools-1.0.7-armv7l-unknown-linux-gnueabihf/lib/udev/\"
armv7l-unknown-linux-gnueabihf-gcc -O2 -Wall -g `pkg-config --cflags uuid blkid`   -c -o bcache.o bcache.c
/nix/store/n313xks5ym0s0a5v8a5285rmnmvy6ms9-bash-4.4-p23/bin/bash: pkg-config: command not found
armv7l-unknown-linux-gnueabihf-gcc -O2 -Wall -g `pkg-config --cflags uuid blkid`    make-bcache.c bcache.o  `pkg-config --libs uuid blkid` -o make-bcache
/nix/store/n313xks5ym0s0a5v8a5285rmnmvy6ms9-bash-4.4-p23/bin/bash: pkg-config: command not found
/nix/store/n313xks5ym0s0a5v8a5285rmnmvy6ms9-bash-4.4-p23/bin/bash: pkg-config: command not found
make-bcache.c:11:10: fatal error: blkid.h: No such file or directory
   11 | #include <blkid.h>
      |          ^~~~~~~~~
compilation terminated.
make: *** [<builtin>: make-bcache] Error 1
builder for '/nix/store/136dpzlngrv5v7z7d7as7k2pj318npwa-bcache-tools-1.0.7-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 2

```

I'm also unsure if we still need the `boot.initrd.extraUdevRulesCommands` / cc @bjornfor 5b48368386864af78658b7c18608df57f8e8a386

cc @NixOS/exotic-platform-maintainers

--- a/pkgs/tools/filesystems/bcache-tools/default.nix
+++ b/pkgs/tools/filesystems/bcache-tools/default.nix
@@ -20,6 +20,7 @@ stdenv.mkDerivation rec {
     sed -e "/INSTALL.*initramfs\/hook/d" \
         -e "/INSTALL.*initcpio\/install/d" \
         -e "/INSTALL.*dracut\/module-setup.sh/d" \
+        -e "s/pkg-config/$PKG_CONFIG/" \
         -i Makefile
   '';
 
@@ -28,8 +29,12 @@ stdenv.mkDerivation rec {
     ./fix-static.patch
   ];
 
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "UDEVLIBDIR=${placeholder "out"}/lib/udev/"
+  ];
+
   preBuild = ''
-    export makeFlags="$makeFlags PREFIX=\"$out\" UDEVLIBDIR=\"$out/lib/udev/\"";
     sed -e "s|/bin/sh|${bash}/bin/sh|" -i *.rules
   '';
 

